lib/finch/http1/conn.ex

```diff
@@ -205,4 +207,9 @@ defmodule Finch.Conn do
         {:error, mint, error}
     end
   end
+
+  defp maybe_log_secrets(scheme, mint_conn) do
+    socket = HTTP1.get_socket(mint_conn)
```

Comment:
Looks like we could just use `HTTP.get_socket/1` for both HTTP1 & HTTP2. Then the implementation would be exactly the same for both